ukvm/ukvm.h

```diff
@@ -142,4 +142,55 @@ extern struct ukvm_module ukvm_module_blk;
 extern struct ukvm_module ukvm_module_net;
 extern struct ukvm_module ukvm_module_gdb;
 
+/*
+ * GDB specific functions to be implemented on all backends for all
+ * architectures.
+ */
+
+/*
+ * Fills *registers with a stream of hexadecimal digits for each register in
+ * GDB register order, where each register is in target endian order.
+ * Returns 0 if success, -1 otherwise.
+ */
+int ukvm_gdb_read_registers(struct ukvm_hv *hv, uint8_t *reg, size_t *len);
+
+/*
+ * Writes the shadow registers from a stream of hexadecimal digits for each register in
+ * GDB register order, where each register is in target endian order.
+ * Returns 0 if success, -1 otherwise.
+ */
+int ukvm_gdb_write_registers(struct ukvm_hv *hv, uint8_t *reg, size_t len);
+
+/*
+ * Enagle single stepping. Returns 0 if success, -1 otherwise.
+ */
+int ukvm_gdb_enable_ss(struct ukvm_hv *hv);
+
+/*
+ * Disable single stepping. Returns 0 if success, -1 otherwise.
+ */
+int ukvm_gdb_disable_ss(struct ukvm_hv *hv);
+
+/*
+ * Reads the current KVM exit code and maps it to a GDB signal value.
+ * Returns 0 if success, -1 otherwise.
+ */
+int ukvm_gdb_read_last_signal(struct ukvm_hv *hv, int *signal);
+
+/*
+ * Add a breakpoint of type software or hardware, at address addr.  len is
```

Comment:
Unclear what are the valid values for `type` here, mention that these are defined in `ukvm_gdb.h`?